src/invariant/InvariantManagerImpl.cpp

```diff
@@ -32,28 +32,32 @@ InvariantManager::create(Application& app)
 }
 
 InvariantManagerImpl::InvariantManagerImpl(medida::MetricsRegistry& registry)
-    : mMetricsRegistry(registry)
+    : mInvariantFailureCount(
+          registry.NewCounter({"ledger", "invariant", "failure"}))
 {
 }
 
 Json::Value
 InvariantManagerImpl::getJsonInfo()
 {
     Json::Value failures;
+
     for (auto const& invariant : mInvariants)
     {
-        auto& counter = mMetricsRegistry.NewCounter(
-            {"invariant", "does-not-hold", "count", invariant.first});
-        if (counter.count() > 0)
+        auto it = mFailureInformation.find(invariant.first);
```

Comment:
I think this loop could be rewritten more simply as
```c++
for (auto const& failure : mFailureInformation)
{
    auto& fail = failures[failure.first];
    auto const& info = failure.second;
    fail["last_failed_on_ledger"] = info.lastFailedOnLedger;
    fail["last_failed_with_message"] = info.lastFailedWithMessage;
}
```

Reply:
good suggestion